bindings/per-isolate-data.cc

```diff
@@ -0,0 +1,37 @@
+#include <unordered_map>
+
+#include "per-isolate-data.hh"
+
+namespace dd {
+
+static std::unordered_map<v8::Isolate*, PerIsolateData*> per_isolate_data_;
+
+PerIsolateData::PerIsolateData(v8::Isolate* isolate) {
+  per_isolate_data_[isolate] = this;
+  node::AddEnvironmentCleanupHook(isolate, [](void* data) {
```

Comment:
I have trouble with understanding the ownership here. 
The object is registered in a static map (but not owned). Why do we call a delete operator on this data ?

Reply:
Ah, should probably be deleted from the map too.

The `node::AddEnvironmentCleanupHook()` function is a helper to run something when Node.js calls `Isolate::Dispose()`. At this point, the `PerIsolateData` for that Isolate is no longer needed as the Isolate is shutting down.

Reply:
Should be fixed now. 😸